CBLClient/Apps/CBLTestServer-Android/app/src/main/java/com/couchbase/CouchbaseLiteServ/server/RequestHandler/ReplicatorConfigurationRequestHandler.java

```diff
@@ -327,4 +362,177 @@ public boolean filtered(Document document, EnumSet<DocumentFlag> flags) {
+        }
+        return localDoc;
+    }
+}
+
+class RemoteWinsCustomConflictResolver implements ConflictResolver {
+    private static final String TAG = "CCRREPLCONFIGHANDLER";
+    @Override
+    public Document resolve(Conflict conflict) {
+        Document localDoc = conflict.getLocalDocument();
+        Document remoteDoc = conflict.getRemoteDocument();
+        String docId = conflict.getDocumentId();
+        String localDocId = localDoc.getId();
+        String remoteDocId = remoteDoc.getId();
+        if (remoteDocId != localDocId) {
+            Log.e(TAG, "Remote docId and local docId are different");
+        }
+        if (remoteDocId != docId) {
+            Log.e(TAG, "Remote docId doesn't match with conflict docId");
+        }
+        if (docId != localDocId) {
+            Log.e(TAG, "Local docId doesn't match with conflict docId");
+        }
+        return remoteDoc;
+    }
+}
+
+class NullCustomConflictResolver implements ConflictResolver {
+    private static final String TAG = "CCRREPLCONFIGHANDLER";
+    @Override
+    public Document resolve(Conflict conflict) {
+        Document localDoc = conflict.getLocalDocument();
+        Document remoteDoc = conflict.getRemoteDocument();
+        String docId = conflict.getDocumentId();
+        String remoteDocId = remoteDoc.getId();
+        String localDocId = localDoc.getId();
+        if (remoteDocId != localDocId) {
+            Log.e(TAG, "Remote docId and local docId are different");
+        }
+        if (remoteDocId != docId) {
+            Log.e(TAG, "Remote docId doesn't match with conflict docId");
+        }
+        if (docId != localDocId) {
+            Log.e(TAG, "Local docId doesn't match with conflict docId");
+        }
+        return null;
+    }
+}
 
+class MergeCustomConflictResolver implements ConflictResolver {
+    private static final String TAG = "CCRREPLCONFIGHANDLER";
+    @Override
+    public Document resolve(Conflict conflict) {
+        Document localDoc = conflict.getLocalDocument();
+        Document remoteDoc = conflict.getRemoteDocument();
+        String docId = conflict.getDocumentId();
+        String remoteDocId = remoteDoc.getId();
+        String localDocId = localDoc.getId();
+        if (remoteDocId != localDocId) {
+            Log.e(TAG, "Remote docId and local docId are different");
+        }
+        if (remoteDocId != docId) {
+            Log.e(TAG, "Remote docId doesn't match with conflict docId");
+        }
+        if (docId != localDocId) {
+            Log.e(TAG, "Local docId doesn't match with conflict docId");
+        }
+        MutableDocument newDoc = localDoc.toMutable();
+        Map<String, Object> remoteDocMap = remoteDoc.toMap();
+        for (Map.Entry<String, Object> entry : remoteDocMap.entrySet()) {
+            String key = entry.getKey();
+            Object value = entry.getValue();
+            if (!newDoc.contains(key)) {
+                newDoc.setValue(key, value);
+            }
+        }
```

Comment:
could you provide some detail of the merge algorithm?

Reply:
Added description as below. Hope this help

/**
         * Migrate the conflicted doc.
         * Algorithm creates a new doc with copying local doc and then adding any additional key
         * from remote doc. Conflicting keys will have value from local doc.
         */